ABCD/data_convert.py

```diff
@@ -0,0 +1,65 @@
+import argparse
+from pathlib import Path
+
+import pandas as pd
+
+
+def get_args():
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description=__doc__)
+    parser.add_argument('-t', '--tabular-data', type=Path, action='store', dest='tab_data', metavar='TAB_DATA_DIR',
+                        help='Path to directory with tabular data files in \'.txt\' format.')
+    parser.add_argument('-d', '--dictionaries', type=Path, action='store', dest='dicts', metavar='DICTS_DIR',
+                        default=Path('.'), help="Path to directory with data dictionaries in \'.csv\' format.")
+    parser.add_argument('-o', '--output-dir', type=Path, action='store', dest='outdir', metavar='OUTPUT_DIR',
+                        default=Path('.'), help="Destination directory for the \'.tsv\' formatted files.")
+    args = parser.parse_args()
+
+    return args.tab_data.resolve(), args.dicts.resolve(), args.outdir.resolve()
+
+
+def data_summary(data_with_no_dict):
+    print(f"A total of {len(data_with_no_dict)} data files don't have a corresponding data dictionary.")
+    print(data_with_no_dict)
+
+
+def main():
+    # get command line arguments
+    tab_data_dir, data_dict_dir, outdir = get_args()
+
+    # extract nda short names into a list
+    data_short_names = [i.name.split('.txt')[0] for i in
+                        tab_data_dir.glob('*.txt')]  # assert that this is indeed a list of strings.
+    data_dicts = list(data_dict_dir.glob('*.csv'))
+    data_files_no_dict = []
+
+    for d in data_short_names:
+        expected_dict = data_dict_dir.joinpath(d + '.csv')  # constructing dictionary filename
+        if expected_dict not in data_dicts:
+            data_files_no_dict.append(d + '.txt')  # constructing data filename
+        else:
+            types_dict = {}  # dict of fields and data types
+            dict_df = pd.read_csv(expected_dict)
+            tsv_fields = dict_df['ElementName'].to_list()
```

Comment:
Oh! Is this the NDA data structure definitions? GitHub username tag me on your response so I can elaborate here.

Reply:
@ericearl what did you want to elaborate on ?

Reply:
It sounds from your above comment that this uses the NDA data structure definition files. So I guess as long as it's clear to the end user they need those in the right place and to make sure they're there before trying this it's probably all good.